src/components/icons/create-styled-icon.js

```diff
@@ -33,29 +33,33 @@ const getSizeStyle = size => {
   }
 };
 
-const getColor = theme => {
-  if (!theme) return 'inherit';
-  switch (theme) {
-    case 'black':
-      return vars.colorSolid;
-    case 'grey':
-      return vars.colorNeutral60;
-    case 'white':
-      return vars.colorSurface;
-    case 'blue':
-      return vars.colorInfo;
-    case 'green':
-      return vars.colorPrimary;
-    case 'green-light':
-      return vars.colorPrimary40;
-    case 'orange':
-      return vars.colorWarning;
-    case 'red':
-      return vars.colorError;
+const getColor = (color, theme) => {
+  if (!color) return 'inherit';
+  const overwrittenVars = {
+    ...vars,
+    ...theme,
+  };
+  switch (color) {
+    case 'solid':
+      return overwrittenVars.colorSolid;
+    case 'neutral60':
+      return overwrittenVars.colorNeutral60;
+    case 'surface':
+      return overwrittenVars.colorSurface;
+    case 'info':
+      return overwrittenVars.colorInfo;
+    case 'primary':
+      return overwrittenVars.colorPrimary;
+    case 'primary40':
+      return overwrittenVars.colorPrimary40;
+    case 'warning':
+      return overwrittenVars.colorWarning;
+    case 'error':
+      return overwrittenVars.colorError;
```

Comment:
Wondering if this could be more of a convention approach instead of switching?

Reply:
Still valid?

Reply:
can you check the latest version?